src/frontend/src/binder/mod.rs

```diff
@@ -116,9 +116,13 @@ pub struct Binder {
 
     param_types: ParameterTypes,
 
-    /// The mapping from sql udf parameters to ast expressions
+    /// The mapping from `sql udf parameters`` to `ast expressions`
     /// Note: The expressions are constructed during runtime, correspond to the actual users' input
     udf_context: HashMap<String, AstExpr>,
+
+    /// The mapping from `function name` to `calling times`
+    /// Record the calling times for every sql udf during the binding phase
+    udf_recursive_context: HashMap<String, i32>,
```

Comment:
Perhaps we should extract them into a struct.

Reply:
> I think we should count the calling depth globally. Otherwise, what if the user constructs a chain:

Actually I don't think this could ever happen, since the prerequisite of such a function calling chain is that we are calling the **existing** sql udf rather than something functions that have not been created. In pg this kind of definition will be rejected during creation time.
reference: https://github.com/risingwavelabs/risingwave/pull/14139#discussion_r1439217732